nncf/common/graph/graph.py

```diff
@@ -594,12 +594,22 @@ def get_graph_for_structure_analysis(self, extended: bool = False) -> nx.DiGraph
             attrs_edge = {}
             u = u.replace(__RESERVED_DOT_CHARACTER, __CHARACTER_REPLACE_TO)
             v = v.replace(__RESERVED_DOT_CHARACTER, __CHARACTER_REPLACE_TO)
+            label = {}
+            if edge[NNCFGraph.PARALLEL_INPUT_PORT_IDS_ATTR]:
+                label["parallel_input_port_ids"] = edge[NNCFGraph.PARALLEL_INPUT_PORT_IDS_ATTR]
+
             if extended:
                 if edge[NNCFGraph.DTYPE_EDGE_ATTR] is Dtype.INTEGER:
                     attrs_edge["style"] = "dashed"
                 else:
                     attrs_edge["style"] = "solid"
-                attrs_edge["label"] = edge[NNCFGraph.ACTIVATION_SHAPE_EDGE_ATTR]
+                label["shape"] = edge[NNCFGraph.ACTIVATION_SHAPE_EDGE_ATTR]
+
+            if label:
+                if len(label) == 1 and extended:
+                    attrs_edge["label"] = label.popitem()[1]
+                else:
+                    attrs_edge["label"] = ", ".join((f"{k}:{v}" for k, v in label.items()))
```

Comment:
Why do we need this?

Reply:
To add information about parallel edges into the dot graph representation. I've added this if to prevent modifications of all dot graph: only graphs with parallel edges information will have additional information in it

Reply:
Can we do it in a more general way? The hard-code like `label.popitem()[1]` looks weird.

Reply:
What would you recommend? is `list(label.values())[0]` looks any better?

Reply:
I recommend reducing hard-coding and implementing it more in a more general way, if possible.

Reply:
Done